pkg/boot/universalpayload/utilities.go

```diff
@@ -170,3 +225,144 @@ func alignHOBLength(expectLen uint64, bufLen int, buf *bytes.Buffer) error {
 	}
 	return nil
 }
+
+// Constrcut trampoline code before jump to entry point of FIT image.
+// Due to lack of support to set value of General Purpose Registers in kexec,
+// bootloader parameter needs to be prepared in trampoline code.
+// Also stack is prepared in trampoline code snippet to ensure no data leak.
+//
+// Trampoline code snippet is prepared as following:
+//
+//	trampoline[0 - 6]   : mov rax, qword ptr [rip+0x19]
+//	trampoline[7 - 9]   : mov rsp, rax
+//	trampoline[10 - 16] : mov rax, qword ptr [rip+0x17]
+//	trampoline[17 - 19] : mov rcx, rax
+//	trampoline[20 - 26] : mov rax, qword ptr [rip+0x15]
+//	trampoline[27 - 28] : jmp rax
+//	trampoline[29 - 31] : padding for alignment
+//	trampoline[32 - 39] : Top of stack address
+//	trampoline[40 - 47] : Base address of bootloader parameter
+//	trampoline[48 - 55] : Entry point of FIT image
+func constructTrampoline(buf []uint8, hobAddr uint64, entry uint64) []uint8 {
+	loadStackAddress := []uint8{0x48, 0x8b, 0x05, 0x19, 0x00, 0x00, 0x00}
+	setStackAddress := []uint8{0x48, 0x89, 0xc4}
+	loadBootparameter := []uint8{0x48, 0x8b, 0x05, 0x17, 0x00, 0x00, 0x00}
+	setBootparameter := []uint8{0x48, 0x89, 0xc1}
+	loadKernelAddress := []uint8{0x48, 0x8b, 0x05, 0x15, 0x00, 0x00, 0x00}
+	jumpToKernelAddress := []uint8{0xff, 0xe0}
+	padForAlignment := []uint8{0x00, 0x00, 0x00}
```

Comment:
You mentioned the necessary instructions above, but having raw bytes is quite opaque and makes maintenance harder. What speaks against doing it like `pkg/boot/multiboot/internal/trampoline/` with go assembly?

Reply:
How about refactor this snippet of opcode to assembly in next PR?

Reply:
Append a new commit convert opcode to golang assembly code, please help review